BuildTools.psm1

```diff
@@ -770,31 +789,26 @@ function Move-TestResults($OutDir) {
 ##############################
 function Run-CasperJs($TestJs='test.js', $Url, [switch]$v11 = $false,
     [string]$OutDir) {
-    $sleepSeconds = 2
-    for ($tryCount = 0; $tryCount -lt 5; $tryCount++) {
-        Start-Sleep -Seconds $sleepSeconds  # Wait for web process to start up.
-        if ($v11) {
-            $env:CASPERJS11_URL = $Url
-            # Casperjs.exe creates a new terminal window, from which we
-            # cannot capture output.  So we use python to invoke it and
-            # capture output.
-            $casperOut = python (Join-Path $env:CASPERJS11_BIN "casperjs") `
-                -- test --xunit=TestResults.xml $TestJs
-            # Casper 1.1 always returns 0, so inspect the xml output
-            # to see if a test failed.
-            [xml]$x = Get-Content TestResults.xml         
-            $LASTEXITCODE = 0      
-            foreach ($suite in $x.testsuites.testsuite) {
-                $LASTEXITCODE += [int] $suite.failures 
-            }
-        } else {
-            $casperOut = casperjs $TestJs $Url
-        }
-        if ($LASTEXITCODE -eq 0) {
-            $casperOut | Write-Host
-            return
+    if ($v11) {
+        $env:CASPERJS11_URL = $Url
+        # Casperjs.exe creates a new terminal window, from which we
+        # cannot capture output.  So we use python to invoke it and
+        # capture output.
+        $casperOut = python (Join-Path $env:CASPERJS11_BIN "casperjs") `
+            -- test --xunit=TestResults.xml $TestJs
+        # Casper 1.1 always returns 0, so inspect the xml output
+        # to see if a test failed.
+        [xml]$x = Get-Content TestResults.xml         
+        $LASTEXITCODE = 0      
+        foreach ($suite in $x.testsuites.testsuite) {
+            $LASTEXITCODE += [int] $suite.failures 
         }
-        $sleepSeconds *= 2
+    } else {
+        $casperOut = casperjs $TestJs $Url
+    }
+    if ($LASTEXITCODE -eq 0) {
+        $casperOut | Write-Host

```

Comment:
Is Casper output tagged somehow so it's easy to recognize that it's casper output?

Reply:
Casper also creates a TestResults.xml, which gets rendered in TestFusion and Sponge.

Reply:
I probably didn't structure the question well. 
I meant if there is any way to distinguish between the traces that are created by casper and the traces that are created by other parts (like dotnet test, ASP.NET service traces, etc)? Assuming that all the traces will be captured in testResults.xml

Also, not relevant for this code review, but in general, in my opinion, it's "dotnet test" that should produce TestResults.xml via custom logger implementation we should supply it. Casper traces should be captured by the test harness (dotnet test) via custom logger.